UPPrayerService/Services/EmailService.cs

```diff
@@ -42,7 +46,74 @@ public async Task SendConfirmationEmail(string recipientEmail, string confirmati
 
             SendGridMessage message = MailHelper.CreateSingleEmail(fromAddress, toAddress, subject, plaintextContent, htmlContent);
 
+
+
+
+            StringBuilder sb = new StringBuilder();
+
+            string Summary = "Pray to End Human Trafficking";
+            string Location = "Anywhere";
+            string Description = "Pray to end human trafficking.";
+            string Organization = "Up Prayer Movement";
+
+            foreach (Models.Reservation reservation in reservations) {
+                //start the calendar item
+                sb.AppendLine("BEGIN:VCALENDAR");
+                sb.AppendLine("VERSION:2.0");
+                sb.AppendLine("PRODID:stackoverflow.com");
+                sb.AppendLine("CALSCALE:GREGORIAN");
+                sb.AppendLine("METHOD:PUBLISH");
+
+                //add the event
+                sb.AppendLine("BEGIN:VEVENT");
+
+                // Create a new date object with the current reservations info.
+                DateTime currentDate = parseDateTimeObject(reservation);
+
+                sb.AppendLine("DTSTART:" + currentDate.ToString("yyyyMMddTHHmm00"));
+                sb.AppendLine("DTEND:" + currentDate.AddMinutes(30).ToString("yyyyMMddTHHmm00"));
+                sb.AppendLine(Guid.NewGuid().ToString());
+
+                sb.AppendLine("SUBJECT:TEST");
+                sb.AppendLine("SUMMARY:" + Summary + "");
+                sb.AppendLine("LOCATION:" + Location + "");
+                sb.AppendLine("DESCRIPTION:" + Description + "");
+                sb.AppendLine("ORGANIZER;CN=" + Organization + ":MAILTO:jsmith@host1.com");
+                sb.AppendLine("PRIORITY:3");
+                sb.AppendLine("END:VEVENT");
+
+                sb.AppendLine("END:VCALENDAR");
+
+                byte[] atachmentBytes = Encoding.ASCII.GetBytes(sb.ToString());
+                string file = Convert.ToBase64String(atachmentBytes);
+
+                message.AddAttachment("invite.ics", file);
+
+                sb.Clear();
+            }
+
             await client.SendEmailAsync(message);
         }
+
+        /*
+         * Takes the reservation info and converts it to a DataTime object.
+        */
+        private DateTime parseDateTimeObject(Models.Reservation reservation)
```

Comment:
Let's refactor this out to the `Models.Reservation` class, and change the name to be more precise.